### PR TITLE
[JIT] Enable addmm fusion for ONNX export only

### PR DESCRIPTION
**Pull request by jamesr66a**
There's some action at a distance issues and not having this is disabling quantization in C2 for prod use cases

ref T34831022

DIRECTLY: After this patch, instead of emitting `Gemm(x, y, 0) + b` in ONNX, we emit `Gemm(x, y, b)`.

### CODE REVIEW

**Comment by jmp84** [comment]
I can confirm the latency regression from https://github.com/pytorch/pytorch/pull/11654 is gone after patching this.

**Review by houseroad — APPROVED** [review]

File: torch/csrc/jit/passes/peephole.cpp [hunk 1]
```diff
@@ -123,8 +122,8 @@ void PeepholeOptimize(Block * block) {
   }
 }
 
-void PeepholeOptimize(std::shared_ptr<Graph>& graph) {
-  PeepholeOptimize(graph->block());
+void PeepholeOptimize(std::shared_ptr<Graph>& graph, bool addmm_fusion_enabled) {
```
Comment (apaszke) [inline torch/csrc/jit/passes/peephole.cpp]: Some comments explaining why this argument exists in the first place would be very useful.

**Comment by ezyang** [comment]
Hi, can we get a unit test for this specific pattern. There is one test affected but it's a big one, I want a small test that zooms in on this particular optimization.

**Comment by dzhulgakov** [comment]
Do you imagine more options being configured? In any case, maybe have an options struct or more semantical variable. Bool arguments can quickly go out of hand.

**Comment by apaszke** [comment]
Yeah a struct only makes sense if we end up having more parameters. There's no risk of boolean blindness right now.

**Review by soumith — APPROVED** [review]

**Pull request closed without merging**